pkg/registry/cluster/registry.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// Registry is an interface implemented by things that know how to store Cluster objects.
+type Registry interface {
+	ListClusters(ctx api.Context, options *api.ListOptions) (*api.ClusterList, error)
+	WatchCluster(ctx api.Context, options *api.ListOptions) (watch.Interface, error)
+	GetCluster(ctx api.Context, clusterID string) (*api.Cluster, error)
```

Comment:
variable name should be name, not clusterID, as name is the primary key of a object in ObjectMeta.